internal/handler/site/middleware/cache.go

```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"bytes"
+	"net/http"
+	"time"
+	"fmt"
+
+	"github.com/oursky/pageship/internal/cache"
+	"github.com/oursky/pageship/internal/httputil"
+	"github.com/oursky/pageship/internal/site"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+
+
+type ContentCacheKey struct {
+	hash        string
+	compression string
+}
+
+type CacheContext struct {
+	cc *cache.ContentCache
+}
+
+func NewCacheContext(cc *cache.ContentCache) CacheContext {
+	return CacheContext{cc: cc}
+}
+
+func (ctx *CacheContext) Cache(site *site.Descriptor, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		info, err := site.FS.Stat(r.URL.Path)
+		if err != nil {
+			http.Error(w, "internal server error", http.StatusInternalServerError)
+			return
+		}
+		
+		keyString := fmt.Sprintf("%v", ContentCacheKey{hash: info.Hash, compression: r.Header.Get("Accept-Encoding")})
```

Comment:
Please construct the key directly; using `%v` format specifier makes the key format unclear. For example, I don't think it would prints unexported fields.